### PR TITLE
feat, refacto(props): implements props data

### PR DESCRIPTION
**Pull request by Prakkkmak**
- Add Prop data
- Refacto
- Create abstract instance creator
- Improve code readibility

### CODE REVIEW

File: code/Modules/Props/Types/CustomProp.cs [hunk 1]
```diff
@@ -1,19 +1,43 @@
...
 
-namespace ssl.Modules.Props
+namespace ssl.Modules.Props.Types
 {
     /// <summary>
     /// A prop is an object not in inventory
     /// Inspired by sandbox Props
     /// </summary>
-    public abstract partial class Prop : BasePhysics, ISelectable
+    public partial class CustomProp : BasePhysics, ISelectable
```
Comment (Nowp) [inline code/Modules/Props/Types/CustomProp.cs:13]: We can use Prop as the class name, we'll just need to make an alias. Prop is cleaner than CustomProp

File: code/Modules/Props/Types/Stain.cs [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
+        }
+
+        public Stain(PropData itemData) : base(itemData)
+        {
+        }
 
         public override void OnAction(MainPlayer player, Item item)
         {
-            if (item is ItemMop)
+            if ("item.mop".Equals(item.Data.Id))
```
Comment (Nowp) [inline code/Modules/Props/Types/Stain.cs:19]: is that a... MAGIC STRING ??

File: config/item.asset [hunk 3]
```diff
@@ -1,8 +1,7 @@
-@BaseClass = item: "An Item"
+@include "data.asset"
+
+@BaseClass base(data) = item : "An item"
 [
-  Id(string) : "Id" : "Item Id"
-  Name(string) : "Name" : "Item Name"
   Description(string) : "Description" : "This is my item"
-  Model(string): "Model" : ""
   HoldType(int): "HoldType" : "1"
```
Comment (Nowp) [inline config/item.asset:6]: We need to use the choices feature to have a dropdown list
syntax: 
	something2(choices) : "second number" : 0 : "Your choice of numbers!" =
	[
		0 : "Default"
		1 : "Something"
		2 : "Another Thing"
	]

source : [https://developer.valvesoftware.com/wiki/FGD](https://developer.valvesoftware.com/wiki/FGD)

**Review by Nowp — CHANGES_REQUESTED** [review]

File: config/item.asset (hunk 3 above)
Comment (Prakkkmak) [reply config/item.asset:6]: To be ticket

**Review by Nowp — APPROVED** [review]

**Pull request merged**